junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/EnabledCondition.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.engine.extension;
+
+import static org.junit.jupiter.api.extension.ConditionEvaluationResult.disabled;
+import static org.junit.jupiter.api.extension.ConditionEvaluationResult.enabled;
+import static org.junit.platform.commons.util.AnnotationUtils.findAnnotation;
+
+import java.lang.reflect.AnnotatedElement;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.script.Bindings;
+import javax.script.ScriptContext;
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+import javax.script.ScriptException;
+
+import org.junit.jupiter.api.Enabled;
+import org.junit.jupiter.api.extension.ConditionEvaluationResult;
+import org.junit.jupiter.api.extension.ExecutionCondition;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.platform.commons.logging.Logger;
+import org.junit.platform.commons.logging.LoggerFactory;
+import org.junit.platform.commons.util.Preconditions;
+
+/**
+ * {@link ExecutionCondition} that supports the {@link Enabled @Enabled} annotation.
+ *
+ * @since 5.1
+ * @see #evaluateExecutionCondition(ExtensionContext)
+ */
+class EnabledCondition implements ExecutionCondition {
+
+	private static final Logger logger = LoggerFactory.getLogger(EnabledCondition.class);
+
+	@Override
+	public ConditionEvaluationResult evaluateExecutionCondition(ExtensionContext context) {
+		Optional<AnnotatedElement> element = context.getElement();
+		Optional<Enabled> annotation = findAnnotation(element, Enabled.class);
+		if (!annotation.isPresent()) {
+			return enabled("@Enabled is not present");
+		}
+
+		// Prepare actual script text from annotation properties
+		Enabled enabled = annotation.get();
+		Preconditions.condition(enabled.value().length > 0, "@Enable#value() array must not be empty");
```

Comment:
FWIW there's a `d` missing.

Reply:
How about `Preconditions.notBlank()`?

Reply:
Not so good. Using `Preconditions.notEmpty` instead. It's an array of `String` (to support multi-line scripts).

Reply:
Ah, sorry, I missed that. If only Java had multiline strings...